File: sample/src/main/java/ru/surfstudio/standard/interactor/common/error/NetworkException.java
package ru.surfstudio.standard.interactor.common.error;

/**
 * базовый класс для всех ошибок, возникающих при работе с сервером
 */
public abstract class NetworkException extends RuntimeException {
    public NetworkException() {
    }

    public NetworkException(String message) {
        super(message);
    }

    public NetworkException(Throwable cause) {
        super(cause);
    }

    public NetworkException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: template/src/main/java/ru/surfstudio/standard/interactor/common/error/NetworkException.kt
package ru.surfstudio.standard.interactor.common.error

/**
 * базовый класс для всех ошибок, возникающих при работе с сервером
 */
abstract class NetworkException : RuntimeException {
    constructor() {}

    constructor(message: String) : super(message) {}

    constructor(cause: Throwable) : super(cause) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}
}
